pkg/controller/webhooks/postgresql.go

```diff
@@ -287,3 +298,52 @@ func validateVacuumRepack(vacuum, repack bool) error {
 	}
 	return nil
 }
+
+func validatePgConf(pg *vshnv1.VSHNPostgreSQL) (fErros field.ErrorList) {
+
+	pgConfBytes := pg.Spec.Parameters.Service.PostgreSQLSettings
+
+	pgConf := map[string]string{}
+	if pgConfBytes.Raw != nil {
+		err := json.Unmarshal(pgConfBytes.Raw, &pgConf)
+		if err != nil {
+			fErros = append(fErros, &field.Error{
+				Field:    "spec.parameters.service.postgresqlSettings",
+				Detail:   fmt.Sprintf("Error parsing pgConf: %s", err.Error()),
+				Type:     field.ErrorTypeInvalid,
+				BadValue: pgConfBytes,
+			})
+			return fErros
+		}
+	}
+
+	blocklist := map[string]string{
```

Comment:
why did you create a map if the values are empty? Please change to a list and move it outside the function at the top of the file. Our code is already very confusing, lets not make it a pain to read.

Reply:
The idea for the map is that we don't have nested loops, so I would keep it as such.

But I agree to move it to the top.

Reply:
You meant that there's no good way to search in the list? I forgot how bad go is in that regard... but there are packages that help with finding an object in the list... I would still have gone with the list... it heart my eyes to see empty strings there, it feels like they will be filled during a for loop....

Reply:
finding an element in array is the same operation in each language, finding key in map is O(1) . O(n^2) You suggest would hurt my eyes ;p